homeworks/franktoth/5_oop/car_fleet_mgr.py

```diff
@@ -0,0 +1,91 @@
+#Creating Car class with the given attributes and methods
+
+class Car:
+    def __init__(self, brand, model, year):
+        self.brand = brand
+        self.model = model
+        self.year = year
+        self.mileage = 0
+        self.fuel_level = 100
+
+    def drive(self, distance):
+        if distance < 0:
+            print(f"Error: Distance cannot be negative.")
+            return
+
+        fuel_consumption = distance * 0.1
+        if fuel_consumption <= self.fuel_level:
+            self.mileage += distance
+            self.fuel_level -= fuel_consumption
+            print(f"Car {self.brand} {self.model} drove {distance} km.")
+        else:
+            possible_distance = self.fuel_level / 0.1
+            self.mileage += possible_distance
+            self.fuel_level = 0
+            print(f"Car {self.brand} {self.model} drove {possible_distance} km and ran out of fuel.")
+
+    def refuel(self, amount):
+        if amount < 0:
+            print("Error: Fuel amount cannot be negative.")
+            return
+        self.fuel_level = min(self.fuel_level + amount, 100)
+        print(f"Car {self.brand} {self.model} refueled. Current fuel level: {self.fuel_level}%")
+
+#Creating Fleet class with the given methods
+
+class Fleet:
+    def __init__(self):
+        self.cars = []
+
+    def add_car(self, car):
+        self.cars.append(car)
+
+    def remove_car(self, brand, model):
+        for car in self.cars:
```

Comment:
Ezt egy picit túlbonyolítod. Elég lenne annyit csinálni hogy:

```
if car in self.cars:
    self.cars.remove(car)
```

Amikor azt nézzük meg hogy "if car in self.cars", akkor az __eq__() method default verziója fut le a háttérben, tehát az objektumok memory addressét fogja megvizsgálni. Mivel ez egyedi minden objektum esetében, így akkor is a megfelelő objektumot fogja törölni ha van 3 Ford Focus a flottában.
Fölösleges tehát megvizsgálni a brandet és a model-t, és loop-olni sem kell.

Ehhez persze a metódusod eredeti verziója kell, amikor a konkrét car objektumot adod át paraméterként:

```
def remove_car(self, car):
    if car in self.cars:
        self.cars.remove(car)
```

így kivéded a ValueError-t, mert csak akkor törölsz amikor biztos vagy benne hogy az a bizonyos objektum ott van a listában.

Reply:
Szia!
Ez volt az eredeti megoldásom, de nem tetszett, mert a funkció használatánál nem éreztem eléggé valóságosnak. Értem, hogy egy user interface-en később meg lehet határozni azt a folyamatot, ami alapján a megfelelő memory addressen lévő object kerül eltávolításra, csak nekem ez túl gépi megoldás volt.
Visszamódosítom akkor az identifer-re az object meghatározását a remove functionon belül. Ahh pedig mennyit melóztam vele :D
Mindjárt megy a változtatás a PR-ba....

Reply:
Azt is megcsinálhatod, hogy minden car objectnek lesz egy rendszáma. Az onnantól tényleg egy egyedi azonosító, és akkor törölhetsz így.
A brand + model nem azonosítja egyedileg az autókat. Viszont akkor már azt is meg kell írni, hogy ne lehessen két ugyanolyan rendszámú autót létrehozni (ezt az init-ben egy egyszerű vizsgálattal el tudod intézni)
Amelyik megoldás szimpatiksuabb :)

Reply:
Szia!
Megcsináltam a rendszámmal. Azért így közelebb van a valósághoz szerintem.
Üdv:
Frank